marimo/_output/formatters/ipython_formatters.py

```diff
@@ -61,6 +68,9 @@ def display(*objs: Any, **kwargs: Any) -> Optional[DisplayHandle]:
                 # Store the object if display_id is provided
                 if display_id is not None:
                     display_objects[display_id] = output_value
+                    # Clean up old display objects if we have too many
+                    if len(display_objects) > 1000:  # Arbitrary limit
+                        clear_display_objects()
 
                 _output.append(output_value)
```

Comment:
Yeah, I think move this line before you set display_objects and you should be fine